### PR TITLE
I2 pay fine control

### PR DESCRIPTION
**Pull request by haynescrossley**
@elliotmitchum PayFineControl class ready for review before merging into master.

Note I hadn't created my branches until today (16/8) so I branched directly of latest master.

### CODE REVIEW

**Comment by elliotmitchum** [comment]
@amrmosit This is ready for your review

File: src/library/payfine/PayFineControl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            throw new RuntimeException("PayFineControl: cannot call setUI except in INITIALISED state");
+        }    
+        this.ui = ui;
+        this.ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void cardSwiped(long patronId) {
+        if (!state.equals(ControlState.READY))
```
Comment (amrmosit) [inline src/library/payfine/PayFineControl.java]: PR44 D1 CSG#7 Curly brackets Must be used

File: src/library/payfine/PayFineControl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    
+    
+    public double payFine(double paymentAmount) {
+        if (!state.equals(ControlState.PAYING)) {
+            throw new RuntimeException("PayFineControl: cannot call payFine except in PAYING state");
+        }
+
+        double change = patron.payFine(paymentAmount);
+        if (change > 0)
+            ui.DiSplAY(String.format("Change: $%.2f", change));
```
Comment (amrmosit) [inline src/library/payfine/PayFineControl.java:53]: PR 44 D2 CSG#7 Curly brackets MUST be used

**Comment by amrmosit** [comment]
@haynescrossley 
Please, there are 2 defects to be reviewed

**Review by amrmosit — COMMENTED** [review]
2 defects to be reviewd

File: src/library/payfine/PayFineControl.java (hunk 1 above)
Comment (haynescrossley) [reply src/library/payfine/PayFineControl.java]: [4f9a9bd](https://github.com/elliotmitchum/ITC205/pull/44/commits/4f9a9bd2d2709cfaafd9f3199a06a8325c12273b)

File: src/library/payfine/PayFineControl.java (hunk 2 above)
Comment (haynescrossley) [reply src/library/payfine/PayFineControl.java:53]: [580f0ab](https://github.com/elliotmitchum/ITC205/pull/44/commits/580f0abc014d30709f653bc1435a07f45e2af065)

**Comment by haynescrossley** [comment]
@amrmosit  resolved changes. Over to you for checking

**Review by amrmosit — APPROVED** [review]
changes have been reviewd. Ready to merge @elliotmitchum 
@haynescrossley

**Comment by elliotmitchum** [comment]
Looks good, PR passes the build checks.

**Pull request merged**